exercises/bank-account/canonical-data.json

```diff
@@ -0,0 +1,317 @@
+            "value": 100
+          },
+          {
+            "operation": "withdraw",
+            "value": 80
+          },
+          {
+            "operation": "withdraw",
+            "value": 20
+          },
+          {
+            "operation": "amount"
+          }
+        ]
+      },
+      "expected": 0
+    },
+    {
+      "uuid": "f9facfaa-d824-486e-8381-48832c4bbffd",
+      "description": "Checking balance of closed account throws error",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "amount"
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
+    },
+    {
+      "uuid": "7a65ba52-e35c-4fd2-8159-bda2bde6e59c",
+      "description": "Deposit into closed account",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "deposit",
+            "value": 50
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
+    },
+    {
+      "uuid": "a0a1835d-faae-4ad4-a6f3-1fcc2121380b",
+      "description": "Deposit into unopened account",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "deposit",
+            "value": 50
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
+    },
+    {
+      "uuid": "570dfaa5-0532-4c1f-a7d3-0f65c3265608",
+      "description": "Withdraw from closed account",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "close"
+          },
+          {
+            "operation": "withdraw",
+            "value": 50
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
+    },
+    {
+      "uuid": "c396d233-1c49-4272-98dc-7f502dbb9470",
+      "description": "Close an account that was not opened",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "close"
+          }
+        ]
+      },
+      "expected": { "error": "account not open" }
+    },
+    {
+      "uuid": "c06f534f-bdc2-4a02-a388-1063400684de",
+      "description": "Open already opened account",
+      "property": "bankAccount",
+      "input": {
+        "operations": [
+          {
+            "operation": "open"
+          },
+          {
+            "operation": "open"
+          }
+        ]
+      },
+      "expected": { "error": "account already open" }
+    },
+    {
+      "uuid": "0722d404-6116-4f92-ba3b-da7f88f1669c",
+      "description": "Reopened account does not retain balance",
```

Comment:
This is a new test case AFAICT, correct? This test case and a0a1835d-faae-4ad4-a6f3-1fcc2121380b are the two test cases of which I'm not sure about. Most existing implementations of this exercise don't have these check I think.

Reply:
Python has both of those tests, I can't speak although for how other tracks have done it.

Reply:
Ah, that's one implementation that I didn't do. Let's keep them then.